packages/kbn-securitysolution-io-ts-list-types/src/common/import_comment/index.test.ts

```diff
@@ -0,0 +1,138 @@
+
+      expect(getPaths(left(message.errors))).toEqual([]);
+      expect(message.schema).toEqual(payload);
+    });
+
+    test('it fails validation when undefined', () => {
+      const payload = undefined;
+      const decoded = importCommentsArray.decode(payload);
+      const message = pipe(decoded, foldLeftRight);
+
+      expect(getPaths(left(message.errors))).toEqual([
+        'Invalid value "undefined" supplied to "Array<(({| comment: NonEmptyString, created_at: string, created_by: string, id: NonEmptyString |} & Partial<{| updated_at: string, updated_by: string, meta: (object | undefined) |}>) | ({| comment: NonEmptyString |} & Partial<{| meta: (object | undefined) |}>))>"',
+      ]);
+      expect(message.schema).toEqual({});
+    });
+
+    test('it fails validation when array includes non ImportComment types', () => {
+      const payload = [1] as unknown as ImportCommentsArray;
+      const decoded = importCommentsArray.decode(payload);
+      const message = pipe(decoded, foldLeftRight);
+
+      expect(getPaths(left(message.errors))).toEqual([
+        'Invalid value "1" supplied to "Array<(({| comment: NonEmptyString, created_at: string, created_by: string, id: NonEmptyString |} & Partial<{| updated_at: string, updated_by: string, meta: (object | undefined) |}>) | ({| comment: NonEmptyString |} & Partial<{| meta: (object | undefined) |}>))>"',
+      ]);
+      expect(message.schema).toEqual({});
+    });
+  });
+
+  describe('importCommentsArrayOrUndefined', () => {
+    test('it passes validation an array of ImportComment', () => {
+      const payload = [...getCommentsArrayMock(), ...getCreateCommentsArrayMock()];
+      const decoded = importCommentsArrayOrUndefined.decode(payload);
+      const message = pipe(decoded, foldLeftRight);
+
+      expect(getPaths(left(message.errors))).toEqual([]);
+      expect(message.schema).toEqual(payload);
+    });
+
+    test('it passes validation when undefined', () => {
+      const payload = undefined;
+      const decoded = importCommentsArrayOrUndefined.decode(payload);
+      const message = pipe(decoded, foldLeftRight);
+
+      expect(getPaths(left(message.errors))).toEqual([]);
+      expect(message.schema).toEqual(payload);
+    });
+
+    test('it fails validation when array includes non ImportComment types', () => {
+      const payload = [1] as unknown as ImportCommentsArrayOrUndefined;
+      const decoded = importCommentsArray.decode(payload);
+      const message = pipe(decoded, foldLeftRight);
+
+      expect(getPaths(left(message.errors))).toEqual([
+        'Invalid value "1" supplied to "Array<(({| comment: NonEmptyString, created_at: string, created_by: string, id: NonEmptyString |} & Partial<{| updated_at: string, updated_by: string, meta: (object | undefined) |}>) | ({| comment: NonEmptyString |} & Partial<{| meta: (object | undefined) |}>))>"',
+      ]);
+      expect(message.schema).toEqual({});
+    });
+  });
+});
```

Comment:
Thanks for the unit tests! They're very helpful for the code review and me testing out `context` and `input`. Appreciate it!